cardinal-components-block/src/main/java/dev/onyxstudios/cca/internal/block/CardinalBlockInternals.java

```diff
@@ -56,6 +56,10 @@ private static synchronized ComponentContainer.Factory<BlockEntity> getBeCompone
             @SuppressWarnings("unchecked") var superclass = (Class<? extends BlockEntity>) entityClass.getSuperclass();
             assert BlockEntity.class.isAssignableFrom(superclass) : "requiresStaticFactory returned false on BlockEntity?";
             factory = /* recursive call */ getBeComponentFactory(superclass);
+
+            // if parent class needs to tick, this one does, too!
+            if(StaticBlockComponentPlugin.INSTANCE.clientTicking.contains(superclass)) StaticBlockComponentPlugin.INSTANCE.clientTicking.add(entityClass);
+            if(StaticBlockComponentPlugin.INSTANCE.serverTicking.contains(superclass)) StaticBlockComponentPlugin.INSTANCE.serverTicking.add(entityClass);
```

Comment:
Are we sure all parent classes went through this method before this class does ?

Reply:
I think so. If you can find a scenario where that isnt the case please let me know.
Anyway I added another test case for it.

Reply:
oh duh, there is the recursive call right above... But, consider a three-levels hierarchy A -> B -> C :
- A has ticking components
- B has non-ticking components
- C has no components for itself

when C loads, it will check only B, not A right ?

Reply:
wait nevermind, B will be considered ticking because the component building method is recursive, so no issue there.